tagExpression/utils.js

```diff
@@ -1,23 +1,143 @@
 const _ = require('lodash');
-const { LexerParser } = require('./LexerParser');
+const { LexerParser, LexerToken } = require('./LexerParser');
+
+/**
+ * Helper method to convert Tag expressions with tag names having special chars to
+ * Lexer Parser safe format, and provide mapping back to tag names after parsing
+ * @param formula - current formula to be parsed
+ * @param tagNames - list of valid existing tags
+ *
+ * @returns {object} -> Contains LexParser safe tag expression and tagToUniqueID mapping
+ *
+ *  { 	
+ * 		originalFormula: string,
+ * 		lexSafeFormula: string,
+ * 		tagNameToUniqueIDMap:Map<string, string>,
+ * 		allTagsValid: boolean	
+ *  }
+ * tagNameToUniqueIDMap : key -> Tag Identifier is the special character trimmed in the original Tag
+ * tagNameToUniqueIDMap : value -> Mapped TagName
+ *
+ * @private
+ */
+function getLexParserSafeTagFormula(formula, tagNames) {
```

Comment:
Same method has been added to common-service. Are you planning to use this method from there?